### PR TITLE
[tutorial] add tictactoc publish script

### PR DESCRIPTION
**Pull request by code-eren**
Add a script for automating the process of publish the tictactoe game contract

### CODE REVIEW

**Comment by code-eren** [comment]
hey @Clay-Mysten @randall-Mysten , very fresh newbie here, it's my first time contributing to this project, trying to get sernse of the contribution workflow, could you approve running it and gave some recommendations? bit confused on whether i am bumping the right version

**Comment by randall-Mysten** [comment]
Hi @code-eren, thanks for your contribution. I will look into this (I am new myself) and follow up. We are actively working on updating the information about how to contribute to Sui. Thanks for your patience!

**Comment by tharbert** [comment]
Hi @code-eren! Thanks a bunch for your work here. I won't comment on the tictactoe script right now however regarding the RELEASES and versioning changes, can you port or cherry-pick `ad6d37e32b557b7c27877839a38c67c886af7d2e` over from the devnet branch? This has the updates including the 0.6.2 release/version references. This is something I should have done myself yesterday during the deployment. Thanks again for contributing.

**Comment by code-eren** [comment]
> Hi @code-eren! Thanks a bunch for your work here. I won't comment on the tictactoe script right now however regarding the RELEASES and versioning changes, can you port or cherry-pick `ad6d37e32b557b7c27877839a38c67c886af7d2e` over from the devnet branch? This has the updates including the 0.6.2 release/version references. This is something I should have done myself yesterday during the deployment. Thanks again for contributing.

Hi @tharbert, cherry-picked. I wonder whether you want this version bump landed in main with this pr, or I can just discard version changes to make it same as current main.

**Review by randall-Mysten — CHANGES_REQUESTED** [review]

File: doc/src/explore/tutorials.md [hunk 1]
```diff
@@ -26,6 +26,41 @@ network will charge for gas using its native currency SUI. This transaction fee
...
+```sh
+#!/bin/bash
+# a bash script to automate the process of publishing the game package
+# this script should be run at root of the repo
+
+# assign address
+CLIENT_ADDRESS=$(sui client addresses | tail -n +2)
+echo $CLIENT_ADDRESS
+ADMIN=$CLIENT_ADDRESS | head -n 1
+PLAYER_X=$CLIENT_ADDRESS | head -n 2
```
[the remark this answers is not in the document]
Comment (code-eren) [reply doc/src/explore/tutorials.md:45]: From my view, they seem use PLAYER_X and PLAYER_O? could you double check?

[the remark this answers is not in the document]
Comment (code-eren) [reply doc/src/explore/tutorials.md]: updated

**Comment by tharbert** [comment]
The releases/versioning looks good to me!

File: doc/src/explore/tutorials.md [hunk 2]
```diff
@@ -22,19 +22,68 @@ Simply leave the terminal with Sui running and start a new terminal for the
...
+network charges for gas using its native currency SUI. This transaction fee
 equals the computational effort of executing operations on the Sui network (i.e.
 gas units) times the price of gas in the SUI currency (i.e. the gas price).
 
+When you complete the setup steps, you can use the either use following script to publish and run the sample code, or perform each step manually. Using the script is optional. To manually run each step, follow the steps starting in the [Gather accounts and gas objects](#gather-accounts-and-gas-objects) section.
+
+## Quick script
+If you prefer not to enter command step by step, or need to go though it multiple
+times for whatever reason (change some move source code ), this automated script
+may be useful to save some time.
```
Comment (randall-Mysten) [inline doc/src/explore/tutorials.md:34]:
```suggestion
If you prefer not to enter command step by step, or need to go though it multiple
times (such as when you change some Move source code), the following automated script
may be useful to save some time.
```

Comment (randall-Mysten) [reply doc/src/explore/tutorials.md:34]: if nothing else, please fix the extra space in `code )`

**Review by randall-Mysten — APPROVED** [review]

**Review by randall-Mysten — APPROVED** [review]

**Review by randall-Mysten — APPROVED** [review]
thanks for the additional commits, sorry for the issues with this

**Comment by Clay-Mysten** [comment]
Interesting addition. Adding Xun, Adam, and Patrick retroactively as an FYI.

**Comment by Clay-Mysten** [comment]
Also wondering whether we could employ this type of script elsewhere.

**Comment by randall-Mysten** [comment]
Xun reviewed the script in Slack prior to merge, just as an FYI

**Comment by Clay-Mysten** [comment]
> Xun reviewed the script in Slack prior to merge, just as an FYI

Ah, thanks Randall and Xun!

**Comment by Clay-Mysten** [comment]
@code-eren , please note we will feature your excellent contribution in our AMA on tech writing tomorrow:
https://discord.com/channels/916379725201563759/925109817834631189/1003637556379910244

**Comment by Clay-Mysten** [comment]
Please join us.

**Pull request merged**